src/ape/api/providers.py

```diff
@@ -1429,6 +1430,12 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
             txn.gas_limit = gas
 
+        elif gas_limit == "max":
+            txn.gas_limit = self.max_gas
```

Comment:
this is a mild change in behavior?

Reply:
without it, i end up in a situation where a txns have none gas, like the validator never runs?
i cant quite figure out why this is happening.
I have added some tests for some ecosystem methods to try and investigate.

As soon as I have a better understanding, I will report back.

I dont think this is really a change in behavior though - more like a bug fix... the problem is that im not sure we should have hit this bug because the gas limit should have gotten set before this point for the local networks. Yet, for some reason in non EIP-1559 networks, it was getting into this state where it skipped paying attention to the network setting....

Ill close this comment with a reasoning as soon as its discovered

Reply:
okay just wanted to double check that it was a bug fix

Reply:
OK - update: The top-level bug in all the L2 plugins.
You can see the fix here: https://github.com/ApeWorX/ape-arbitrum/pull/20

This fix is still nice because it handles the case where you are calling methods in a weird one (kinda anti-normal flow).
See my new test in `test_provider` to see what i mean, the setup of the test.

However, the biggie here is that I need to update all the L2s that dont use EIP-1559 to handle gas limits appropriately in their `create_transaction()` override.